ocaml/xapi/taskHelper.ml

```diff
@@ -279,7 +277,7 @@ let failed ~__context exn =
         Db_actions.DB_Action.Task.set_backtrace ~__context ~self
           ~value:(Sexplib.Sexp.to_string Backtrace.(sexp_of_t (get exn))) ;
         Db_actions.DB_Action.Task.set_finished ~__context ~self
-          ~value:(Date.of_float (Unix.time ())) ;
+          ~value:(Date.now ()) ;
```

Comment:
Unfortunately both this and Unix.time only measure seconds, and lose fractional seconds. We should probably fix that independent of this PR. (Unix.gettimeofday () would've been better).

I don't know why we do so many conversions in the Date module, but there is a lot of tricky backward compatibility code there with various parts of the code expecting the date in a certain format. We should probably store the full precision timestamp, and only drop precision when printing in the backward compatible formats.

Anyway something to solve independent of this PR.

Reply:
Date.now uses Ptime_clock.now, which claims nanosecond precision:

```
#install_printer Ptime.Span.pp;;
# Ptime_clock.period () ;;
- : Ptime.span option = Some 1ns
# let ptime_pp = Ptime.pp_human ~frac_s:6 ();;
val ptime_pp : Format.formatter -> Ptime.t -> unit = <fun>
# #install_printer ptime_pp;;
# Ptime_clock.now () ;;
- : Ptime.t = 2024-07-09 13:20:17.875620 -00:00
```

Reply:
> . (Unix.gettimeofday () would've been better).
I don't remember why does date convert the nanosecon-precision timestamp from ptime to integer seconds, but the internal representation of date should change, as you say

Reply:
I believe these "dates" should be serialized to the database as timestamps, not human-readable dates